tests/models/whisper/test_modeling_whisper.py

```diff
@@ -1406,6 +1406,56 @@ def test_tiny_specaugment_librispeech(self):
         # fmt: on
         self.assertTrue(torch.allclose(logits[0][0, 0, :30].cpu(), EXPECTED_LOGITS, atol=1e-4))
 
+    @slow
+    def test_generate_with_prompt_ids(self):
+        processor = WhisperProcessor.from_pretrained("openai/whisper-tiny")
+        model = WhisperForConditionalGeneration.from_pretrained("openai/whisper-tiny")
+        model.to(torch_device)
+        input_speech = self._load_datasamples(4)[-1:]
+        input_features = processor(input_speech, return_tensors="pt").input_features
+
+        output_without_prompt = model.generate(input_features)
+        prompt_ids = processor.get_prompt_ids("Leighton")
+        output_with_prompt = model.generate(input_features, prompt_ids=prompt_ids)
+
+        expected_without_prompt = "<|startoftranscript|><|en|><|transcribe|><|notimestamps|> He has grave doubts whether Sir Frederick Layton's work is really Greek after all and can discover in it but little of Rocky Ithaca.<|endoftext|>"
+        expected_with_prompt = "<|startofprev|> Leighton<|startoftranscript|><|en|><|transcribe|><|notimestamps|> He has grave doubts whether Sir Frederick Leighton's work is really Greek after all and can discover in it but little of Rocky Ithaca.<|endoftext|>"
+        self.assertEqual(processor.decode(output_without_prompt[0]), expected_without_prompt)
+        self.assertEqual(processor.decode(output_with_prompt[0]), expected_with_prompt)
+
+    @slow
+    def test_generate_with_prompt_ids_and_forced_decoder_ids(self):
```

Comment:
I feel like this is most readable / simplest as one test with comments clarifying the cases, lmk if you want them split into separate unit tests

Reply:
I agree it's very readable but there's a potential issue: the `model` will keep state around, i.e. it will change the `model.generation_config` object with the new `forced_decoder_ids` and this may affect the next test. So I think it's better to instantiate a new model before each test.

Maybe it's also a good idea to test what happens after you do the following, just to make sure the code can handle both of these things being None:

```python
model.config.forced_decoder_ids = None
model.generation_config.forced_decoder_ids = None
```

Reply:
Added a case for the above which involved a change in `generate` I'll call out below. I was aiming to order the tests to prevent conflicting state issues but you're right they're more brittle that way, split them into individual tests

Reply:
Sorry, moving parts, we had a test explicitly for this when there we 5 test cases. Then we trimmed them, per this https://github.com/huggingface/transformers/pull/22496#discussion_r1188856955 I changed the `test_generate_with_prompt_ids_and_no_non_prompt_forced_decoder_ids` test to use `whisper-base.en` and `return_timestampt=True`. I just tested it tho and realized that combination didn't actually set those attributes to `None`, so I updated the test to explicity set those two to `None`.

tl;dr it was tested, then wasn't, now is again